### ISSUE TITLE
twisted.web.http_headers.Headers doesn't case normalize headers passed to its initializer

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3557]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3557|
|Type|enhancement|
|Created|2008-11-28 20:18:44Z|
|Branch|https://github.com/twisted/twisted/tree/headers-initializer-3557|

This is surprising and inconvenient:

```
>>> from twisted.web.http_headers import Headers
>>> h = Headers({'Foo': ['bar']})
>>> h.getRawHeaders('foo')
>>> 
```

The result should be `['bar']`, by lower-casing all the headers passed to `__init__`.

<details><summary>Searchable metadata</summary>

```
trac-id__3557 3557
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_headers_initializer_3557 branches/headers-initializer-3557
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1227903524000000 1227903524000000
changetime__1229353228000000 1229353228000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [25580]) Branching to 'headers-initializer-3557'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @exarkun|
|-|-|

Please merge!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [25667]) Merge headers-initializer-3557

Author: exarkun
Reviewer: therve
Fixes: [#3557](https://github.com/twisted/twisted/issues/3557)

Change the `twisted.web.http_headers.Headers` initializer so that it normalizes
the case of the raw header names passed to it.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
